### ISSUE TITLE
[TABS]: Keyboard navigation issues in Safari & FireFox

### ISSUE THREAD

**Issue by ginabak** [Shopify/polaris#3830]
## Issue summary

During the 🎩 of https://github.com/Shopify/web/pull/36467 **Bump polaris v5.13.0**, I noticed the following issues:

1. Keyboard navigation in `Tabs` seems to be problematic in Safari & FireFox (see GIFS and expand to see behaviour)
2. Safari: Cannot easily tab into any of the tabs via the keyboard
3. Firefox: Cannot tab into disclosure portion via the keyboard

## Expected behavior
One should be able to navigate tabs easily via the Keyboard. See w3 example [here](https://www.w3.org/TR/wai-aria-practices/examples/disclosure/disclosure-navigation.html). 
<details><summary> Expand for GIF </summary> 
![Chrome-Proper-Tabs](https://user-images.githubusercontent.com/43223543/104487401-0bba9080-559b-11eb-9895-b8cebb8dec4f.gif)
</details>

## Actual behavior

<details><summary> Expand for GIFs </summary>

### SAFARI BEHAVIOUR
![Safari-Tab-Issue](https://user-images.githubusercontent.com/43223543/104486657-0d378900-559a-11eb-9ce7-fa80a5a99b3d.gif)

### FIREFOX BEHAVIOUR
![Firefox-Tab-issues](https://user-images.githubusercontent.com/43223543/104486669-13c60080-559a-11eb-930f-112c8dbfdd9a.gif)

</details>

## Steps to reproduce the problem

(set up local environment) 
#### Safari behaviour
1. Open Safari
2. Go to https://shop1.myshopify.io/admin/products 
3. Ensure you have a disclosure, like this: 
<img width="715" alt="Screen Shot of Tabs with Disclosure Option" src="https://user-images.githubusercontent.com/43223543/104486996-8a62fe00-559a-11eb-884b-1328050da231.png">
4. Try to tab via the keyboard

#### FireFox behaviour
1. Open FireFox
2. Go to https://shop1.myshopify.io/admin/products 
3. Ensure you have a disclosure, like this: 
<img width="715" alt="Screen Shot of Tabs with Disclosure Option" src="https://user-images.githubusercontent.com/43223543/104486996-8a62fe00-559a-11eb-884b-1328050da231.png">
4. Try to tab via the keyboard

**Comment by metamoni** [comment]
I've tested the Tab component both on Safari and Firefox, and you can navigate through tabs by using the Left/Right Arrow keys. This seems to be the expected behaviour. [This article by Heydon Pickering](https://inclusive-components.design/tabbed-interfaces/) is a great resource for learning more about making tabs accessible.

https://user-images.githubusercontent.com/22390758/134815894-1830b278-42e0-4735-a6bd-561c41df9565.mov

**Comment by aaronbnb** [comment]
**Issue summary**
I am following up here. This issue was first reported 14 months ago and still remains. 

The [Tabs navigation](https://polaris.shopify.com/components/navigation/tabs#navigation) does not conform to WAI-ARIA patterns. This issue is true for all web browsers - Safari, Firefox, and Chrome.

@metamoni you shared a helpful article with Shopify. That article demonstrates that the current implementation actually does not match the expected behavior for tabs - in numerous ways.

**Actual Behavior**
* When one presses the right or left cursor key, focus is incorrectly set to the tab panel. Strangely, a person must press the Tab or Shift+Tab key to return focus to the currently selected tab. 

* That Tab or Shift+Tab returns focus to the currently selected tab does not make sense. A page's natural focus order is set by the sequence of HTML source code - unless positive tabindex attributes or JavaScript are used to actively add focus management and change the natural tab order. A logical focus order goes from top to bottom. When one is in the tab panel, they are further down the page from the set of tabs yet when they press Tab, they're moved back in the page's focus order.

* In smaller viewports, the Tab panel condenses and has a kebab menu button which expands to reveal more tabs. When this button is activated, focus is set to the first tab in the expander. This is inappropriate. The menu button has an aria-expanded attribute. When you move focus from the button, the change in the aria-expanded attribute is not announced to screen readers. 

**Expected Behavior**
* For navigating/selecting tabs, the correct pattern, depends on whether Shopify wants the tabs to be manually or automatically activated. It'd be great if Shopify could provide both options. For automatic activation, the expected behavior is that a person presses tab to sets focus to the currently selected tab and then presses the right/left cursor keys to change the tab selection (back and forth) without again engaging the Tab key. For manual activation, pressing cursor keys should set focus to the neighboring tab and then pressing the Tab key should move to the page's next focusable element after the set of tabs.

* For the kebab menu button, which reveals more tabs, the implementation, which would conform to WAI-ARIA and general screen reader navigation patterns, is to keep focus on the button so that the screen reader announces the change from collapsed to expanded. 

I'm happy to provide further feedback.

**Resources**
[WAI-ARIA Authoring Practices 1.2 Tabs with automatic activation](https://w3c.github.io/aria-practices/examples/tabs/tabs-1/tabs.html)
[WAI-ARIA Authoring Practices 1.2 Tabs with manual activation](https://www.w3.org/TR/wai-aria-practices-1.2/examples/tabs/tabs-2/tabs.html)
